### PR TITLE
Added analytic event triggers for Lookup Dictionary

### LINKED ISSUE #8263: Add Analytic event for "Lookup Dictionary" (see if we can remove it)
I don't think the feature is used. It's 10 years old, esoteric, hard to access, and probably deprecated by the share/system context menu.

It adds complexity to `AbstractFlashcardViewer`, and may be better as a separate app to allow system-wide usage, instead of an AnkiDroid specific feature.

I would like an analytic event to see if users still use it, to argue for its removal in 2.16

Likely entry point:
https://github.com/ankidroid/Anki-Android/blob/a079a61fc0e7dc06437c851026b29266868aef9b/AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java#L78-L83

### PR DESCRIPTION
**Pull request by ritvij14**
## Purpose / Description
Adds analytic event triggers to `Lookup.java` to understand frequency of user interaction with Lookup Dictionary.

## Fixes
Fixes #8263

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by ritvij14** [comment]
@david-allison-1 needed some help in testing out the changes. How do I access the development version of google analytics for this project?

File: AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java [hunk 1]
```diff
@@ -83,11 +82,13 @@ public static boolean lookUp(String text) {
         text = text.trim().replaceAll("[,;:\\s(\\[)\\].]*$", "").replaceAll("^[,;:\\s(\\[)\\].]*", "");
         switch (mDictionary) {
             case DICTIONARY_NONE:
+                UsageAnalytics.sendAnalyticsEvent(Lookup.class.getSimpleName(), "none");
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java:85]: Could you extract the string constants to `UsageAnalytics`, as done in https://github.com/ankidroid/Anki-Android/pull/8257/files#diff-9c4a73a86961824e0bd5352bf8625b7599134dbfdfe1f39f343c0ca8f2a3eb10R360

**Review by david-allison-1 — COMMENTED** [review]
Looks good! Minor refactoring and good to go

**Comment by david-allison-1** [comment]
> @david-allison-1 needed some help in testing out the changes. How do I access the development version of google analytics for this project?

We'll need to discuss that, for now, assume that if it logs the value, it works

**Comment by ritvij14** [comment]
> > @david-allison-1 needed some help in testing out the changes. How do I access the development version of google analytics for this project?
> 
> We'll need to discuss that, for now, assume that if it logs the value, it works

Yes, after some modifications, I can see them in logcat

This is log on selecting ColorDict/GoldenDict
```
2021-03-20 22:59:59.164 24539-24539/com.ichi2.anki D/ViewRootImpl: [TouchInput][ViewRootImpl] MotionEvent { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=53501656, downTime=53501656 } moveCount:0
2021-03-20 22:59:59.280 24539-24539/com.ichi2.anki D/ViewRootImpl: [TouchInput][ViewRootImpl] MotionEvent { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=53501772, downTime=53501656 } moveCount:0
2021-03-20 22:59:59.309 24539-24624/com.ichi2.anki D/Surface: Surface::disconnect(this=0x7eefbc6000,api=1)
2021-03-20 22:59:59.312 24539-24624/com.ichi2.anki D/OpenGLRenderer: endAllActiveAnimators on 0x7e96e3a400 (AlertController$RecycleListView) with handle 0x7ef43e0220
2021-03-20 22:59:59.313 24539-24539/com.ichi2.anki D/View: [Warning] assignParent to null: this = DecorView@9dc712c[Preferences]
2021-03-20 22:59:59.325 24539-24539/com.ichi2.anki D/UsageAnalytics: sendAnalyticsEvent() category/action/value/label: Lookup/colordict/-2147483648/null
2021-03-20 22:59:59.326 24539-24539/com.ichi2.anki D/UsageAnalytics: getOptIn() status: true
```

**Comment by david-allison-1** [comment]
I don't quite understand the changes here - we've moved the analytics from usage of the feature to the toggling of the feature within preferences?

Won't this mean that users that have already toggled the preference won't be included in analytics?

**Comment by ritvij14** [comment]
@david-allison-1 I think I got confused in the usage of `Lookup.java`, sorry about that 😅. Do we need analytics only for the usage of the dictionary or for both usage + toggling?

**Comment by ritvij14** [comment]
@david-allison-1 I checked for only usage of dictionary and yes the analytics are working

```
2021-03-22 09:46:06.997 19103-19103/com.ichi2.anki D/ViewRootImpl: [TouchInput][ViewRootImpl] MotionEvent { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=120229845, downTime=120229845 } moveCount:0
2021-03-22 09:46:07.027 19103-19103/com.ichi2.anki D/ViewRootImpl: [TouchInput][ViewRootImpl] MotionEvent { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=120229878, downTime=120229845 } moveCount:0
2021-03-22 09:46:07.050 19103-19103/com.ichi2.anki I/AbstractFlashcardViewer: AbstractFlashcardViewer:: Lookup button pressed
2021-03-22 09:46:07.061 19103-19103/com.ichi2.anki I/Timeline: Timeline: Activity_launch_request time:120229914
2021-03-22 09:46:07.101 19103-19103/com.ichi2.anki D/UsageAnalytics: sendAnalyticsEvent() category/action/value/label: Lookup/eijiro/-2147483648/null
2021-03-22 09:46:07.101 19103-19103/com.ichi2.anki D/UsageAnalytics: getOptIn() status: true
```
Please review the PR.

**Comment by david-allison-1** [comment]
> @david-allison-1 I think I got confused in the usage of `Lookup.java`, sorry about that 😅. Do we need analytics only for the usage of the dictionary or for both usage + toggling?

No worries! Let's just stick to usage of the feature for now

File: AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java [hunk 2]
```diff
@@ -115,25 +115,30 @@ public static boolean lookUp(String text) {
                             mLookupText = "";
                         })
                         .build().show();
+                UsageAnalytics.sendAnalyticsEvent(Lookup.class.getSimpleName(), UsageAnalytics.Actions.LEO);
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java:118]: We may need to split `Leo` into Web and App, but that'll only be worth it if we find it has usages. Looks good for now, feel free to mark as 'resolved' with no changes

**Review by david-allison-1 — APPROVED** [review]
Looks good to me! Thanks so much for your patience

File: AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java (hunk 2 above)
Comment (ritvij14) [reply AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java:118]: @david-allison-1 actually, looking back at the code, maybe we can add the analytic triggers in this part separately for web and app?
https://github.com/ankidroid/Anki-Android/blob/5ead42cd0ebd0319731757ec91e821498067a387/AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java#L143-L160

Or should I leave it as it is for now?

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java:118]: I'd be fine with leaving it, 

But if you'd prefer to fix this, it feels like a good idea to keep all the logs in the same method - it'd be better to split out the 'case' statements

**Comment by david-allison-1** [comment]
We'll squash and merge this PR when it's merged.

For the future, it'd be great if this could be done via an interactive `git rebase` on your end.

**Review by mikehardy — APPROVED** [review]
Nice!

**Comment by mikehardy** [comment]
Hi there!

Just a friendly notice (one per person for PRs merged in March, regardless of PR quantity) that we try to process OpenCollective payments monthly - it's time for March 2021 submissions

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

Thanks!

**Pull request merged**